Tactics/Reflect/AxEffects.lean

```diff
@@ -359,6 +395,28 @@ private def assertIsDefEq (e expected : Expr) : MetaM Unit := do
   if !(←isDefEq e expected) then
     throwError "expected:\n  {expected}\nbut found:\n  {e}"
 
+/-- Given an expression `e : ArmState`,
+which is a sequence of `w`/`write_mem`s to `eff.currentState`,
+return an `AxEffects` where `e` is `currentState`.
+
+Note that as soon as an unsupported expression (e.g., an `if`) is encountered,
+the whole expression is taken to be the initial state,
```

Comment:
I know this line is from before, but it could be made more accurate. Right now, it seems as though this is an assumption but isn't an error thrown if the expressions isn't def-eq to the initial state?
```
the whole expression is taken to be the initial state,
```

Reply:
You're right, this line is only true for `fromExpr`, not for `updateExpr` (we check that the expression is def-eq the the previous *currentState*, but same difference)